skbio/alignment/_tabular_msa.py

```diff
@@ -1701,7 +1701,15 @@ def reassign_index(self, mapping=None, minter=None):
                 "Cannot use both `mapping` and `minter` at the same time.")
 
         if mapping is not None:
-            self._seqs.rename(mapping, inplace=True)
+            if isinstance(mapping, dict):
+                self.index = [mapping[label] if label in mapping else label
```

Comment:
Okay, awesome. Thank you. I love the ternary
On May 4, 2016 1:17 PM, "Evan Bolyen" <notifications@github.com> wrote:

> In skbio/alignment/_tabular_msa.py
> <https://github.com/biocore/scikit-bio/pull/1325#discussion_r62106829>:
>
> > @@ -1701,7 +1701,15 @@ def reassign_index(self, mapping=None, minter=None):
> >                  "Cannot use both `mapping` and `minter` at the same time.")
> >
> >          if mapping is not None:
> > -            self._seqs.rename(mapping, inplace=True)
> > +            if isinstance(mapping, dict):
> > +                self.index = [mapping[label] if label in mapping else label
>
> Developer preference I think. My preference probably matches the ternary
> as it looks a little "prettier" and it's purpose feels clearer in this
> context. But there's nothing wrong with your approach either.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1325/files/0183591e24cca3243bebd454fb76ef2519933c58..a988e7faf40030a92e90dac3f4f0518e10ea2849#r62106829>
>